src/chains/ethereum/transaction/src/eip1559-fee-market-transaction.ts

```diff
@@ -122,7 +122,14 @@ export class EIP1559FeeMarketTransaction extends RuntimeTransaction {
       this.accessListJSON = accessListData.AccessListJSON;
       this.validateAndSetSignature(data);
     }
-    this.updateEffectiveGasPrice();
+    // there's a chance this value was already set from the `extra` passed to
+    // the base transaction, and we don't want to overwrite it
+    if (
+      this.effectiveGasPrice === undefined ||
+      this.effectiveGasPrice.valueOf() === undefined
+    ) {
+      this.updateEffectiveGasPrice();
+    }
```

Comment:
Maybe we should call `this.updateEffectiveGasPrice()` at the end of `autofillDefaultTransactionValues` instead of here. It seems out of place check these values here, as `maxFeePerGas`, which `updateEffectiveGasPrice` uses internally, might not be set yet (or `Quantity.isNull() === true`). I could be wrong though :sweat_smile: